TestCommon/net/ports_* failure on UC16

As pointed out in https://github.com/canonical/edgex-checkbox-provider/pull/57#issuecomment-1424464182, the following tests fail on Ubuntu Core 16:
- TestCommon/net/ports_not_listening_on_all_interfaces
- TestCommon/net/ports_listening_on_localhost

```
=== RUN   TestCommon
=== RUN   TestCommon/net
=== RUN   TestCommon/net/ports_open
    net.go:140: Retry 1/60: Waiting for ports: 59880 (core-data), 59881 (core-metadata), 59882 (core-command), 8500 (consul), 6379 (redis)
=== CONT  TestCommon/net
    net.go:140: Retry 1/60: Waiting for ports: 59880 (core-data), 59881 (core-metadata), 59882 (core-command), 8500 (consul), 6379 (redis)
=== RUN   TestCommon/net/ports_not_listening_on_all_interfaces
    exec.go:19: [exec] sudo lsof -nPi :59880 || true
    net.go:268: Port 59880 is not open
=== RUN   TestCommon/net/ports_listening_on_localhost
    exec.go:19: [exec] sudo lsof -nPi :59880 || true
    net.go:268: Port 59880 is not open
=== RUN   TestCommon/query_readings
    readings.go:36: waiting for device-virtual to produce readings, current retry count: 1/60
    readings.go:39: device-virtual is producing readings now, readings queried from core-data
--- FAIL: TestCommon (1.02s)
    --- FAIL: TestCommon/net (0.02s)
        --- PASS: TestCommon/net/ports_open (0.00s)
        --- FAIL: TestCommon/net/ports_not_listening_on_all_interfaces (0.01s)
        --- FAIL: TestCommon/net/ports_listening_on_localhost (0.00s)
    --- PASS: TestCommon/query_readings (1.00s)
```

The ports are checked and open right before this failure. This may be an error from the `sudo lsof -nPi :59880` command, rather than a problem with the services.

--- a/data/latest/run-go-tests.sh
+++ b/data/latest/run-go-tests.sh
@@ -35,6 +35,9 @@ print_logs() {
 }
 trap print_logs EXIT
 
+# Use the staged snap command available to the user
+sed -i 's/sudo snap/snap/g' ./test/utils/snap.go
+
 # TODO:
 sed -i '/TestTLSCert/a t.Skip("https://github.com/canonical/edgex-checkbox-provider/issues/52")' ./test/suites/edgexfoundry/proxy_test.go
 sed -i '/TestAddProxyUser/a t.Skip("https://github.com/canonical/edgex-checkbox-provider/issues/55")' ./test/suites/edgexfoundry/proxy_test.go

--- a/data/latest/run-go-tests.sh
+++ b/data/latest/run-go-tests.sh
@@ -35,8 +35,10 @@ print_logs() {
 }
 trap print_logs EXIT
 
-# Use the staged snap command available to the user
-sed -i 's/sudo snap/snap/g' ./test/utils/snap.go
+# The sudoer's path (secure_path) misses $SNAP/usr/bin/ which we need for
+# commands such as: snap, lsof. Remove all sudo commands and run everything as
+# current user:
+find test -type f -exec sed -i 's/sudo //g' {} +
 
 # TODO:
 sed -i '/TestTLSCert/a t.Skip("https://github.com/canonical/edgex-checkbox-provider/issues/52")' ./test/suites/edgexfoundry/proxy_test.go

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -92,12 +92,12 @@ parts:
     organize:
       '*': bin/
   testing-tools:
-      plugin: nil
-      stage-packages:
-        - git # path and config set in run-go-tests.sh
-        - golang-1.18-go # PATH set in run-go-tests.sh
-        # - libcurl4-openssl-dev
-        # - curl # this uses libcurl3. adding it causes libssl errors in openssl commands
-        # - openssl
-        # - libssl-dev
-      
+    plugin: nil
+    stage-packages:
+      - git # path and config set in run-go-tests.sh
+      - golang-1.18-go # PATH set in run-go-tests.sh
+      - snapd
+      # - libcurl4-openssl-dev
+      # - curl # this uses libcurl3. adding it causes libssl errors in openssl commands
+      # - openssl
+      # - libssl-dev

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -92,12 +92,12 @@ parts:
     organize:
       '*': bin/
   testing-tools:
-    plugin: nil
-    stage-packages:
-      - git # path and config set in run-go-tests.sh
-      - golang-1.18-go # PATH set in run-go-tests.sh
-      - snapd
-      # - libcurl4-openssl-dev
-      # - curl # this uses libcurl3. adding it causes libssl errors in openssl commands
-      # - openssl
-      # - libssl-dev
+      plugin: nil
+      stage-packages:
+        - git # path and config set in run-go-tests.sh
+        - golang-1.18-go # PATH set in run-go-tests.sh
+        - snapd
+        # - libcurl4-openssl-dev
+        # - curl # this uses libcurl3. adding it causes libssl errors in openssl commands
+        # - openssl
+        # - libssl-dev

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -95,8 +95,11 @@ parts:
       plugin: nil
       stage-packages:
         - git # path and config set in run-go-tests.sh
-        - golang-1.18-go # PATH set in run-go-tests.sh
-        - snapd
+        - golang-1.18-go # path set in run-go-tests.sh
+        # The checkbox-edgexfoundry snap based on core20 expects the snap binary 
+        # at a path that isn't compatible with how snapd is installed on UC16.
+        # See: https://github.com/canonical/edgex-checkbox-provider/issues/59
+        - snapd # re-install the snap command
         # - libcurl4-openssl-dev
         # - curl # this uses libcurl3. adding it causes libssl errors in openssl commands
         # - openssl

